example/spec/scaling/scalingSpec.js

```diff
@@ -0,0 +1,115 @@
+    AutoScalingGroupName: autoScalingGroupName,
+    MaxRecords: 1
+  };
+  let activties = await autoscaling().describeScalingActivities(params).promise();
+  const startingActivity = activties.Activities[0];
+  let mostRecentActivity = Object.assign({}, startingActivity);
+  /* eslint-disable no-await-in-loop */
+  while (startingActivity.ActivityId === mostRecentActivity.ActivityId) {
+    activties = await autoscaling().describeScalingActivities(params).promise();
+    mostRecentActivity = activties.Activities[0];
+    console.log(`No new activity found. Sleeping for ${waitPeriod / 1000} seconds.`);
+    await sleep(waitPeriod);
+  }
+  /* eslint-enable no-await-in-loop */
+
+  return mostRecentActivity;
+}
+
+const numExecutions = 3;
+const numActivityTasks = 1;
+const minInstancesCount = 1;
+
+describe('When a task is configured to run in Docker', () => {
+  beforeAll(async () => {
+    clusterArn = await getClusterArn(config.stackName);
+    autoScalingGroupName = await getAutoScalingGroupName(config.stackName);
+  });
+
+  describe('the load on the system exceeds that which its resources can handle', () => {
+    let workflowExecutionArns = [];
+
+    beforeAll(async () => {
+      const workflowExecutionPromises = [];
+      for (let i = 0; i < numExecutions; i += 1) {
+        workflowExecutionPromises.push(buildAndStartWorkflow(
+          config.stackName,
+          config.bucket,
+          workflowName
+        ));
+      }
+      workflowExecutionArns = await Promise.all(workflowExecutionPromises);
+    });
+
+    afterAll(async () => {
+      const completions = workflowExecutionArns.map((executionArn) => waitForCompletedExecution(executionArn));
+      await Promise.all(completions);
+    });
+
+    it('can handle the load', async () => {
+      await sleep(5000);
+      const stats = await getClusterStats({});
+      expect(stats.runningEC2TasksCount + stats.pendingEC2TasksCount).toEqual(numExecutions + numActivityTasks);
+    });
+
+    it('adds new resources', async () => {
+      console.log('Waiting for scale out policy to take affect.');
+      const mostRecentActivity = await getNewScalingActivity();
+      expect(mostRecentActivity.Description).toMatch(/Launching a new EC2 instance: i-*/);
+    });
+  });
+
+  describe('the load on the system is far below what its resources can handle', () => {
+    it('removes excessive resources but not all resources', async () => {
+      console.log('Waiting for scale in policy to take affect.');
+      const mostRecentActivity = await getNewScalingActivity();
+      expect(mostRecentActivity.Description).toMatch(/Terminating EC2 instance: i-*/);
+      const stats = await getClusterStats({});
+      expect(stats.runningEC2TasksCount + stats.pendingEC2TasksCount).toEqual(numActivityTasks);
+      const instances = await ecs().listContainerInstances({ cluster: clusterArn }).promise();
+      expect(instances.containerInstanceArns.length).toEqual(minInstancesCount);
```

Comment:
Is it only possible to predict the amount of container instances when the system is at minimal load?

Reply:
We could write an expectation for the number of container instances when at maximum load. Above you might notice I'm testing instead there exists an auto scaling activity of `Launching a new EC2 instance`. I decided to use the new scaling activity itself as a test the policy is working as expected. I realize the extra wait time is inconsequential to total test time seeing as the test waits for the instance to be destroyed as well. I think a test which waits for the instance to be in service _could_ be a little more flaky than testing for the scaling activity since the instance could be in service for only a short time before being terminated by the scale in/down policy.

I think testing a new instance vs testing the scaling activity is taking place are effectively the same test but I can add a test for the number of instances in service. What do you think @markdboyd @bilts ?

Reply:
I think what you're saying and what you're already doing makes sense, but honestly I'm a bit out of my depth here with ECS. So I defer to @bilts here